src/utils/cookieJar.js

```diff
@@ -1,15 +1,137 @@
-/*
-Copyright 2019 Adobe. All rights reserved.
-This file is licensed to you under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License. You may obtain a copy
-of the License at http://www.apache.org/licenses/LICENSE-2.0
+/**
+ MIT License
 
-Unless required by applicable law or agreed to in writing, software distributed under
-the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
-OF ANY KIND, either express or implied. See the License for the specific language
-governing permissions and limitations under the License.
-*/
+ Copyright (c) 2018 Copyright 2018 Klaus Hartl, Fagner Brack, GitHub Contributors
 
-import cookie from "@adobe/reactor-cookie";
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
 
-export default cookie;
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ */
+
+// All cookies we write will be coming from server response bodies and their
+// names and values will have already been properly encoded for cookie storage.
+// It's important that we do not attempt to re-encode this data. By default,
+// the @adobe/reactor-cookie utility provided by Launch encodes names and values.
+// @adobe/reactor-cookie exposes `get`, `set`, and `remove` methods from the
+// underlying js-cookie utility (Launch didn't export more of js-cookie in order
+// to reduce the amount of API Launch needed to maintain). In order to modify
+// encoding behavior, however, @adobe/reactor-cookie would need to export the
+// `withConverter` method of js-cookie. However, even then, `withConverter` will
+// only allow us to modify encoding behavior of cookie values and not cookie
+// names. This was submitted as an issue to js-cookie here:
+// https://github.com/js-cookie/js-cookie/issues/560
+// Until this gets resolved, we need to roll our own or use some other library
+// that allows us to avoid encoding and decoding.
```

Comment:
This is important to understand. Currently I haven't added tests for this module. The content of this module comes from js-cookie and any encoding or decoding has been stripped out. My plan is to create a pull request against js-cookie that allows us to essentially disable encoding/decoding, but I don't know if they'll accept it and, even then, I don't know that Launch will want to expose it (that's more API Launch has to maintain, making it more difficult to switch implementations later on if necessary). If all those things happen, then we can probably just use `@adobe/reactor-cookie`. In the meantime, and in case those things don't happen, I also plan on creating a separate package called js-cookie-without-encoding that will be a fork of the js-cookie repo but trimmed down to the code in this module. I plan on publishing it to npm and then we can use js-cookie-without-encoding here.